Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp

```diff
@@ -88,84 +95,111 @@ namespace Multiplayer
-            console->PerformCommand(loadLevelString.c_str());
+        // Setup the normal multiplayer connection.
+        // This needs to be done before in-memory spawnable creation and level loading
+        // because the entity alias resolution is dependent on connection type
+        AZ::Interface<IMultiplayer>::Get()->InitializeMultiplayer(MultiplayerAgentType::DedicatedServer);
+        INetworkInterface* networkInterface = AZ::Interface<INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName));
 
-            // Setup the normal multiplayer connection
-            AZ::Interface<IMultiplayer>::Get()->InitializeMultiplayer(MultiplayerAgentType::DedicatedServer);
-            const AZ::Name mpNetworkInterfaceName = AZ::Name(MpNetworkInterfaceName);
-            INetworkInterface* networkInterface = AZ::Interface<INetworking>::Get()->RetrieveNetworkInterface(mpNetworkInterfaceName);
+        // Create in-memory spawnables and load dependent assets. This ensures dependent spawnables are loaded
+        // when the level is loaded by path. Otherwise the dependent spawnables may not load because they will
+        // already have a "Ready" status from the AssignAssetData call in InMemorySpawnableAssetContainer
+        // (ex. root.spawnable depends on the in-memory root.network.spawnable)
+        constexpr bool loadDependentAssets = true;
+        AzFramework::InMemorySpawnableAssetContainer::CreateSpawnableResult createSpawnableResult =
+            m_inMemorySpawnableAssetContainer->CreateInMemorySpawnableAsset(rootSpawnableAssetDataInfoContainer, loadDependentAssets, "Root");
+        if (!createSpawnableResult.IsSuccess())
+        {
+            AZ_Assert(false, "MultiplayerEditorConnection failed to create level spawnable. Error result: %s", createSpawnableResult.GetError().c_str())
+        }
 
-            uint16_t sv_port = DefaultServerPort;
-            if (console->GetCvarValue("sv_port", sv_port) != AZ::GetValueResult::Success)
-            {
-                AZ_Assert(false,
-                    "MultiplayerEditorConnection::HandleRequest for EditorServerLevelData failed! Could not find the sv_port cvar; we won't be able to listen on the correct port for incoming network messages! Please update this code to use a valid cvar!")
-            }
-            
-            networkInterface->Listen(sv_port);
+        // Spawnable library needs to be rebuilt since now we have newly registered in-memory spawnable assets
+        AZ::Interface<INetworkSpawnableLibrary>::Get()->BuildSpawnablesList();
 
-            AZLOG_INFO("Editor Server completed receiving the editor's level assets, responding to Editor...");
-            return connection->SendReliablePacket(MultiplayerEditorPackets::EditorServerReady());
+        // Load the level via the root spawnable that was registered
+        const auto console = AZ::Interface<AZ::IConsole>::Get();
+        console->PerformCommand("LoadLevel Root.spawnable");
```

Comment:
This seems like doing it the long way around. Isn't there a way to directly access the level system?

Reply:
Direct call uses gEnv->pSystem->GetILevelSystem() so it'll require setting up multiplayer module to listen for crysystem events and defining gEnv. Whatcha think? Is it better bringing in old gEnv or using the console? I'm cool either way, this is only called once per CTRL+G so performance isn't a worry. Console would save a few lines of code.